tools/desktop/desktop.go

```diff
@@ -0,0 +1,234 @@
+			Usage:   "Version of the Fleet Desktop application",
+			EnvVars: []string{"FLEET_DESKTOP_VERSION"},
+		},
+		&cli.BoolFlag{
+			Name:    "verbose",
+			Usage:   "Log detailed information when building the application",
+			EnvVars: []string{"FLEET_DESKTOP_VERBOSE"},
+		},
+	}
+
+	app.Commands = []*cli.Command{
+		macos(),
+	}
+	return app
+}
+
+func macos() *cli.Command {
+	return &cli.Command{
+		Name:        "macos",
+		Usage:       "Creates the Fleet Desktop Application for macOS",
+		Description: "Builds and signs the Fleet Desktop .app bundle for macOS",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "authority",
+				Usage:   "Authority to use on the codesign invocation (if not set, app is not signed)",
+				EnvVars: []string{"FLEET_DESKTOP_APPLE_AUTHORITY"},
+			},
+		},
+		Action: func(c *cli.Context) error {
+			if !c.Bool("verbose") {
+				zlog.Logger = zerolog.Nop()
+			}
+			return createMacOSApp(c.String("version"), c.String("authority"))
+		},
+	}
+}
+
+func createMacOSApp(version, authority string) error {
+	const (
+		appDir           = "Fleet Desktop.app"
+		bundleIdentifier = "com.fleetdm.desktop"
+		// infoPList is the Info.plist file to use for the macOS .app bundle.
+		//
+		// 	- NSHighResolutionCapable=true: avoid having a blurry icon and text.
+		//	- LSUIElement=1: avoid showing the app on the Dock.
+		infoPList = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleExecutable</key>
+	<string>fleet-desktop</string>
+	<key>CFBundleIdentifier</key>
+	<string>%s</string>
+	<key>CFBundleInfoDictionaryVersion</key>
+	<string>6.0</string>
+	<key>CFBundleName</key>
+	<string>fleet-desktop</string>
+	<key>CFBundlePackageType</key>
+	<string>APPL</string>
+	<key>CFBundleShortVersionString</key>
+	<string>%s</string>
+	<key>CFBundleVersion</key>
+	<string>%s</string>
+	<key>NSHighResolutionCapable</key>
+	<string>True</string>
+	<key>LSUIElement</key>
+	<string>1</string>
+</dict>
+</plist>
+`
+	)
+
+	if runtime.GOOS != "darwin" {
+		return errors.New(`the "Fleet Desktop" macOS app can only be created from macOS`)
+	}
+
+	defer os.RemoveAll(appDir)
+
+	contentsDir := filepath.Join(appDir, "Contents")
+	macOSDir := filepath.Join(contentsDir, "MacOS")
+	if err := secure.MkdirAll(macOSDir, constant.DefaultDirMode); err != nil {
+		return fmt.Errorf("create directory %q: %w", macOSDir, err)
+	}
+
+	infoFile := filepath.Join(contentsDir, "Info.plist")
+	infoPListContents := fmt.Sprintf(infoPList, bundleIdentifier, version, version)
+	if err := ioutil.WriteFile(infoFile, []byte(infoPListContents), 0o644); err != nil {
+		return fmt.Errorf("create Info.plist file %q: %w", infoFile, err)
+	}
+
+	/* #nosec G204 -- arguments are actually well defined */
+	buildExec := exec.Command("go", "build", "-o", filepath.Join(macOSDir, "fleet-desktop"), "./"+filepath.Join("orbit", "cmd", "desktop"))
```

Comment:
Perhaps this?
```suggestion
	buildExec := exec.Command("go", "build", "-o", filepath.Join(macOSDir, "fleet-desktop"), filepath.Join(".", "orbit", "cmd", "desktop"))
```

Reply:
Didn't help, looks like the only thing that would work is replacing the `filepath.Join` calls with the actual strings.